src/Orleans/Utils/DictionaryExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orleans
+{
+    static class DictionaryExtensions
+    {
+        public static TResult Find<TKey, TResult>(this IDictionary<TKey, TResult> dictionary, TKey key) where TResult : class
```

Comment:
In that case one formulation, thinking this project and #209 would be
> Do provide comments on functions. Exception is non-public surface when the comments would state the obvious. Or public, parameterless, default constructors which would state ``//This is a public default constructor``.

(I need to be away for some hours now, in case we continue this dialog.)